spacy/cli/debug_data.py

```diff
@@ -247,6 +255,69 @@ def debug_data(
                     msg.warn(f"No examples for texts WITHOUT new label '{label}'")
                     has_no_neg_warning = True
 
+            with msg.loading("Obtaining span characteristics..."):
+                span_characteristics = _get_span_characteristics(
+                    train_dataset, gold_train_data, spans_key
+                )
+
+            msg.info(f"Span characteristics for spans_key `{spans_key}`")
+            msg.info("SD = Span Distinctiveness, BD = Boundary Distinctiveness")
+            _print_span_characteristics(span_characteristics)
+
+            _span_freqs = _get_spans_length_freq_dist(
+                gold_train_data["spans_length"][spans_key]
+            )
+            _filtered_span_freqs = _filter_spans_length_freq_dist(
+                _span_freqs, threshold=SPAN_LENGTH_THRESHOLD_PERCENTAGE
+            )
+
+            msg.info(
+                f"At least {SPAN_LENGTH_THRESHOLD_PERCENTAGE}% of spans have up to "
+                f"{max(_filtered_span_freqs.keys())} n-grams "
+                f"(min={span_characteristics['min_length']}, max={span_characteristics['max_length']}). "
+                f"Most common span lengths are: {_format_freqs(_filtered_span_freqs)}. "
+                "If you are using the n-gram suggester, note that omitting "
+                "infrequent n-gram lengths can greatly improve speed and "
+                "memory usage with only a small effect on recall."
```

Comment:
I don't like my own edit!

```suggestion
                "memory usage."
```